autocertkit/ssh.py

```diff
@@ -179,71 +178,67 @@ def __init__(self,
                             nowarn=nowarn)
         self.command = command
         self.nolog = nolog
-        self.ignoreExitCode = ignoreExitCode
-        if string.find(command, "\n") > -1 and not newlineok:
-            log.debug("Command with newline: '%s'" % (command))
+
         try:
             self.client = self.open_session()
             self.client.settimeout(timeout)
-            self.client.set_combine_stderr(True)
+            self.client.set_combine_stderr(combineStderr)
             self.client.exec_command(command)
             self.client.shutdown(1)
-            self.fh = self.client.makefile()
+            self.hStdout = self.client.makefile()
+            self.hStderr = None if combineStderr else self.client.makefile_stderr()
         except Exception, e:
             self.reply = "SSH connection failed",
             self.toreply = 1
             self.close()
 
-    def read(self, retval="code", fh=None):
-        """Process the output and result of the command.
-
-        @param retval: Whether to return the result code (default) or 
-            stdout as a string.
-
-            string  :   Return a stdout as a string.
-            code    :   Return the result code. (Default). 
-
-            If "string" is used then a failure results in an exception.
-
-        """
-
-        if self.toreply:
-            if retval == "string":
-                raise Exception(self.reply)
-            return self.reply
+    def read_file(self, inf, outf=None, label="stdout"):
```

Comment:
add doc string to illustrate how to use the function.

Reply:
Needless I think, there is doc string for the original interface, see below 'read' function.
In my opinion, we should keep the "Style" during refactor, unless we modify all.